### ISSUE TITLE
`Model.logpdf` incorrectly broadcasts over arrays with many parameter configurations

### ISSUE THREAD

**Issue by phinate** [scikit-hep/pyhf#1893]
### Summary

When evaluating the `logpdf` of a pyhf model, I would naively expect that I could feed in a series of parameters `[[a1, b1],[a2,b2],...[aN,bN]]` and get a result of shape `(N, 1)`, but instead I get something that sums over the whole parameter grid.

From debugging this, I found that it seems to be localised to the machinery in `parameters/paramview.py`, e.g. in the [calculation of `self.all_indicies` here](https://github.com/scikit-hep/pyhf/blob/86229ad3257ddbbdb78548d7ceade5c9ab683a0c/src/pyhf/parameters/paramview.py#L35-L52) which is `[0,1]`, and is seemingly then used to access individual indicies of the parameter array, which is actually totally flattened [here](https://github.com/scikit-hep/pyhf/blob/86229ad3257ddbbdb78548d7ceade5c9ab683a0c/src/pyhf/parameters/paramview.py#L103).

I wasn't able to debug further since this part of the codebase is hard to follow. Example of this below:

### OS / Environment

```console
MacBook Air 2020 [M1]
Software:

    System Software Overview:

      System Version: macOS 12.4 (21F79)
      Kernel Version: Darwin 21.5.0
      Time since boot: 24 days 16:55
```

### Steps to Reproduce

```python (paste below)
import pyhf
import numpy as np
pyhf.set_backend("jax")

def make_model(s, b, bup, bdown):
    return pyhf.simplemodels.correlated_background(s, b, bup, bdown)

s = [1,2,3]
b = [21,20,22]

bup = [22,22,24]
bdown = [20, 18, 20]

model1 = make_model(s, b, bup, bdown)
data = [23, 23, 23] + model1.config.auxdata

grid = (500, 500)
x_range = np.linspace(0,10,grid[0])
y_range = np.linspace(0,10,grid[1]) 
xx, yy = np.meshgrid(x_range, y_range)
xy = np.concatenate([xx.reshape(-1, 1), yy.reshape(-1, 1)], axis=1)
def logpdf(pars):
    return model1.logpdf(pars, data)
print(xy.shape, logpdf(xy))
#> (250000, 2) [-9.63806898]
# expected result of shape (250000, 1)
```

### Expected Results

Expected a broadcasting over a long vector of parameters to return a result of the same first dimension.

### Actual Results

```console
Outputs array of shape (1).
```

### pyhf Version

```console
0.6.4.dev149 [installed from git, commit 86229ad3257ddbbdb78548d7ceade5c9ab683a0c]
```

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by phinate** [comment]
If using the `jax` backend, there's a super-easy workaround with `jax.vmap`, here in the same example:

```python3
@jax.vmap
def logpdf(pars):
    return model1.logpdf(pars, data)[0]
logpdf(jnp.array(xy)).shape
#> (25000,)
```

**Comment by alexander-held** [comment]
I am not sure that it is related, but perhaps you are meant to set `batch_size` ahead of time for the model (#545)?

**Comment by phinate** [comment]
ah, I forgot about that functionality! thanks for the pointer :)
